### PR TITLE
clean up secret generators

### PR DESCRIPTION
**Pull request by kbhawkey**
- Clean up content about `secretGenerators` (related to PR #17642).
- Started content cleanup (capitalization, clarity).
- This page is quite dense. In the future, possibly split up page.

### CODE REVIEW

**Comment by kbhawkey** [comment]
@mikedanese , as the original author, would you have time to review? Thank you.

**Comment by kbhawkey** [comment]
Page preview:
https://deploy-preview-18320--kubernetes-io-master-staging.netlify.com/docs/concepts/configuration/secret/

File: content/en/docs/concepts/configuration/secret.md [hunk 1]
```diff
@@ -25,78 +24,92 @@ is safer and more flexible than putting it verbatim in a
...
 secret.
 
 The automatic creation and use of API credentials can be disabled or overridden
-if desired.  However, if all you need to do is securely access the apiserver,
+if desired.  However, if all you need to do is securely access the API server,
 this is the recommended workflow.
 
-See the [Service Account](/docs/tasks/configure-pod-container/configure-service-account/) documentation for more
-information on how Service Accounts work.
+See the [Service Account](/docs/tasks/configure-pod-container/configure-service-account/)
```
Comment (sftim) [inline content/en/docs/concepts/configuration/secret.md]: One optional tweak if you think it helps:
```suggestion
See the [ServiceAccount](/docs/tasks/configure-pod-container/configure-service-account/)
```
(and maybe write “service account” in all-lowercase on the next line?

Comment (kbhawkey) [reply content/en/docs/concepts/configuration/secret.md]: 👍

**Comment by sftim** [comment]
/lgtm

These changes make the page better than it was. Nice work @kbhawkey and @PabloCastellano

**Pull request merged**